button-merchant/src/main/java/com/usebutton/merchant/AndroidEncoder.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * AndroidEncoder.java
+ *
+ * Copyright (c) 2019 Button, Inc. (https://usebutton.com)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+package com.usebutton.merchant;
+
+import android.util.Base64;
+
+/**
+ * Encoder implementation for the Android platform.
+ *
+ * @see Encoder
+ */
+public class AndroidEncoder implements Encoder {
+    @Override
+    public String encodeBase64ToString(byte[] bytes) {
+        return Base64.encodeToString(bytes, Base64.NO_WRAP);
```

Comment:
We can't test this due to the fact that `Base64` is an Android specific class. In fact, I abstracted it out into an interface specifically due to that fact because then I can mock this behavior and test everything else around it.

We also don't need to test this class because it is part of the API provided by Android and thus falls out of our _coverage_.

Reply:
This is exactly why code coverage is not easy to measure unless we can ignore files that include Android specific dependencies.

Unfortunately, it does not seem this is possible with coveralls.

https://github.com/kt3k/coveralls-gradle-plugin/issues/62